### PR TITLE
[incubator/jaeger] fix hotrod demo application

### LINKED ISSUE #10137: [incubator/jaeger] hotrod component configured with wrong environment variable: JAEGER_AGENT_HOST_PORT
**Is this a request for help?**:
No
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Bug report

**Version of Helm and Kubernetes**:
```
helm version
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
**Which chart**:
`incubator/jaeger`

**What happened**:
After helm install, hotrod is failing. K8s logs from hotrod pod:
```
2018-12-19T14:01:53.701Z	INFO	cmd/root.go:84	Using expvar as metrics backend
2018-12-19T14:01:53.701Z	INFO	cmd/all.go:27	Starting all services
2018-12-19T14:01:53.701Z	FATAL	log/logger.go:47	cannot parse Jaeger env vars	{"service": "customer", "error": "cannot obtain reporter config from env: cannot parse env var JAEGER_AGENT_PORT=udp://10.3.0.120:5775: strconv.ParseInt: parsing \"udp://10.3.0.120:5775\": invalid syntax", "errorVerbose": "strconv.ParseInt: parsing \"udp://10.3.0.120:5775\": invalid syntax\ncannot parse env var JAEGER_AGENT_PORT=udp://10.3.0.120:5775\ngithub.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/config.reporterConfigFromEnv\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/config/config_env.go:179\ngithub.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/config.FromEnv\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/config/config_env.go:86\ngithub.com/jaegertracing/jaeger/examples/hotrod/pkg/tracing.Init\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/examples/hotrod/pkg/tracing/init.go:32\ngithub.com/jaegertracing/jaeger/examples/hotrod/cmd.glob..func2\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/examples/hotrod/cmd/customer.go:39\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333\ncannot obtain reporter config from env\ngithub.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/config.FromEnv\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/config/config_env.go:89\ngithub.com/jaegertracing/jaeger/examples/hotrod/pkg/tracing.Init\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/examples/hotrod/pkg/tracing/init.go:32\ngithub.com/jaegertracing/jaeger/examples/hotrod/cmd.glob..func2\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/examples/hotrod/cmd/customer.go:39\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333"}
github.com/jaegertracing/jaeger/examples/hotrod/pkg/log.logger.Fatal
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/examples/hotrod/pkg/log/logger.go:47
github.com/jaegertracing/jaeger/examples/hotrod/pkg/tracing.Init
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/examples/hotrod/pkg/tracing/init.go:34
github.com/jaegertracing/jaeger/examples/hotrod/cmd.glob..func2
	/home/travis/gopath/src/github.com/jaegertracing/jaeger/examples/hotrod/cmd/customer.go:39
```

**What you expected to happen**:
hotrod example app should be running successfully.

**How to reproduce it** (as minimally and precisely as possible):
Here's my helm command:
```
helm upgrade --install jaeger incubator/jaeger \                                                                                                              --namespace monitoring \
    --set provisionDataStore.cassandra=false \
    --set provisionDataStore.elasticsearch=false \
    --set storage.type=memory \
    --set hotrod.enabled=true
```

**Anything else we need to know**:

### LINKED ISSUE #10138: [incubator/jaeger] no traces appearing from hotrod example app
**Is this a request for help?**:
yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
bug report

**Version of Helm and Kubernetes**:
```
helm version
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:
`incubator/jaeger`

**What happened**:
Related to [10137](https://github.com/helm/charts/issues/10137). After issue 10137, I downloaded the chart and added the following env. var to hotrod-deploy.yaml:
```
          env:
            - name: JAEGER_AGENT_HOST
              value: "jaeger-agent.monitoring"
            - name: JAEGER_AGENT_PORT
              value: "6831"
```
After helm install, all pods are now running. When I browse to hotrod and generate some traffic no traces are available from jaeger web ui.

**What you expected to happen**:
If my fix corrected hotrod then I'd expect some traces available from jaeger web ui.

**How to reproduce it** (as minimally and precisely as possible):
1. apply above fix to hotrod-deploy.yaml
2. helm install from downloaded chart
3. generate some traffic

**Anything else we need to know**:
The only errors I see are:
From query:
```
kubectl logs jaeger-query-f5bf69987-bk9jh                                                                                                                1646ms  Wed 19 Dec 14:31:52 2018
{"level":"info","ts":1545228531.167349,"caller":"healthcheck/handler.go:99","msg":"Health Check server started","http-port":16687,"status":"unavailable"}
{"level":"info","ts":1545228531.1722057,"caller":"memory/factory.go:55","msg":"Memory storage configuration","configuration":{"MaxTraces":0}}
{"level":"info","ts":1545228531.1723406,"caller":"query/main.go:186","msg":"Archive storage not created","reason":"Archive storage not supported"}
{"level":"info","ts":1545228531.3670552,"caller":"query/main.go:134","msg":"Registering metrics handler with HTTP server","route":"/metrics"}
{"level":"info","ts":1545228531.3671126,"caller":"healthcheck/handler.go:133","msg":"Health Check state change","status":"ready"}
{"level":"info","ts":1545228531.367125,"caller":"query/main.go:143","msg":"Starting HTTP server","port":16686}
{"level":"error","ts":1545228832.172297,"caller":"zap/logger.go:33","msg":"error when flushing the buffer: write udp 127.0.0.1:37256->127.0.0.1:6831: write: connection refused","stacktrace":"github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/log/zap.(*Logger).Error\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/log/zap/logger.go:33\ngithub.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go.(*remoteReporter).processQueue.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/reporter.go:257\ngithub.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go.(*remoteReporter).processQueue\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/uber/jaeger-client-go/reporter.go:267"}
```

From hotrod:
```
...
2018-12-19T14:13:48.301Z	INFO	log/spanlogger.go:34	Dispatch successful	{"service": "frontend", "driver": "T725910C", "eta": "2m0s"}
2018-12-19T14:13:48.353Z	ERROR	log/spanlogger.go:39	redis timeout	{"service": "driver", "driver_id": "T733351C", "error": "redis timeout"}
2018-12-19T14:13:48.353Z	ERROR	log/spanlogger.go:39	Retrying GetDriver after error	{"service": "driver", "retry_no": 1, "error": "redis timeout"}
2018-12-19T14:13:48.428Z	ERROR	log/spanlogger.go:39	redis timeout	{"service": "driver", "driver_id": "T758618C", "error": "redis timeout"}
2018-12-19T14:13:48.428Z	ERROR	log/spanlogger.go:39	Retrying GetDriver after error	{"service": "driver", "retry_no": 1, "error": "redis timeout"}
2018-12-19T14:13:48.463Z	INFO	log/spanlogger.go:34	Search successful	{"service": "driver", "num_drivers": 10}
...
```

### PR DESCRIPTION
**Pull request by sebastien-prudhomme**
Signed-off-by: Sébastien Prud'homme <sebastien.prudhomme@gmail.com>

#### What this PR does / why we need it:

Hotrod demo application not working with this chart

#### Which issue this PR fixes

  - fixes #10137, fixes #10138

fix also the name of the jaeger agent service

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: incubator/jaeger/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 1.8.2
 description: A Jaeger Helm chart for Kubernetes
 name: jaeger
-version: 0.8.0
+version: 0.8.1
```
Comment (pavelnikolov) [inline incubator/jaeger/Chart.yaml]: Version `0.8.1` has already been [used](https://github.com/helm/charts/blob/master/incubator/jaeger/Chart.yaml#L5). 
Please, bump the version

**Review by pavelnikolov — CHANGES_REQUESTED** [review]
LGTM. Just fix the version, please.

File: incubator/jaeger/Chart.yaml (hunk 1 above)
Comment (sebastien-prudhomme) [reply incubator/jaeger/Chart.yaml]: Done

**Pull request merged**